staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor.go

```diff
@@ -107,24 +107,36 @@ func (c *convertor) ConvertToTable(ctx context.Context, obj runtime.Object, tabl
 				continue
 			}
 
-			// as we only support simple JSON path, we can assume to have only one result (or none, filtered out above)
-			value := results[0][0].Interface()
-			if customHeaders[i].Type == "string" {
-				if err := column.PrintResults(buf, []reflect.Value{reflect.ValueOf(value)}); err == nil {
-					cells = append(cells, buf.String())
-					buf.Reset()
-				} else {
-					cells = append(cells, nil)
-				}
+			headerType := customHeaders[i].Type
+			if headerType == "string" || len(results[0]) > 1 {
+				// Use string value if header type is string, but also if there are multiple results
+				// because cellForStringValue() can return a string representation of an array of values
+				cells = append(cells, cellForStringValue(column, results))
 			} else {
-				cells = append(cells, cellForJSONValue(customHeaders[i].Type, value))
+				cells = append(cells, cellForJSONValue(customHeaders[i].Type, results[0][0].Interface()))
```

Comment:
Note that this change needs to verify that kubectl and other client ecosystem tools gracefully handle arrays, but i would probably expect clients to handle arrays (we left the door open for that).  That said, if clients behave poorly, we might want to clarify that you can use one or the other.

Reply:
I would generally expect to keep data in the JSON form as it goes across the wire - collapsing to a string is something a client should do based on its context.

Reply:
Actually another PR was just merged (https://github.com/kubernetes/kubernetes/pull/89660#event-3283101863) and it affected the output, so I had to rebase and upate my tests.  I guess it is a little more JSON-like now:
`[1 2] [3]` --> `[1,2] [3]`
`[foo bar] [baz]` --> `["foo","bar"] ["baz"]`

Currently, without this PR, the client just gets an incomplete result (first element only) if there is more than one element, so even if not entirely JSON-parse-able, it is still better than it was.

Let me know if you would like any other tweaks to the output though, to make it better.

Reply:
Disregard my above comment.  I just pushed another update that has what I think you requested... Table Converter does not do any string formatting and leaves the types alone going over the wire, and Table Printer has been updated to output slices in JSON format.

Reply:
@smarterclayton can you take another look at this PR?  I revised it to only add unit tests documenting/showing the current behavior for now, with the next step (a different PR perhaps) being to figure out how to change the code to produce a new desired output.  Thanks!